pyquil/api/_qam.py

```diff
@@ -42,16 +41,12 @@ class QAM(ABC):
     pretend to be a QPI-compliant quantum computer.
     """
 
-    status: str
-    _client: Client
-    _variables_shim: Dict[ParameterAref, Union[int, float]]
-    executable: Optional[QuantumExecutable]
-    experiment: Optional[Experiment] = None
-    _memory_results: Dict[str, Optional[np.ndarray]]
```

Comment:
I think as long as you annotate them properly in `__init__()`, then the type information is available. The variables declared just under `class` will still be class variables, but they can be overridden/shadowed as instance variables if set on `self`. This, in my recent experience, has led to some strange errors when you try to set default values that are intended to be instance variables. If the code that overrides/shadows them is never executed, then the value is the same for all instances. At first, you'd think this wouldn't be a problem, since the default value is the same for all instances, but it gets tricky with data types that are more complex than a simple int or string. For instance, if you have a dictionary that is set to `{}` by default in the `class` area outside of `__init__()`, then all instances end up updating a shared dictionary (I ran into this situation with the `EngagementManager` and it was not apparent what was going on at first).

Thus, I opted to avoid them altogether and stuck with pure instance members defined in `__init__()`. I think the code is less readable, but it'll be easier to debug.